core/src/serde_support.rs

```diff
@@ -0,0 +1,1454 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::format;
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
+use base64::Engine;
+use dusk_bytes::Serializable;
+use serde::de::{
+    EnumAccess, Error as SerdeError, MapAccess, Unexpected, VariantAccess,
+    Visitor,
+};
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::stake::{Reward, RewardReason, SlashEvent, StakeEvent, StakeKeys};
+use crate::transfer::withdraw::WithdrawReceiver;
+use crate::transfer::{
+    ContractToAccountEvent, ContractToContractEvent, ConvertEvent,
+    DepositEvent, MoonlightTransactionEvent, PhoenixTransactionEvent,
+    WithdrawEvent,
+};
+use crate::{
+    signatures::bls::PublicKey as AccountPublicKey, stake::StakeFundOwner,
+};
+use crate::{BlsScalar, String};
+
+// To serialize and deserialize u64s as big ints:
+// https://github.com/dusk-network/rusk/issues/2773#issuecomment-2519791322.
+struct Bigint(u64);
```

Comment:
Please also add some tests for the `Bigint` serialization as well, especially edge-cases like `u64::MAX`

Reply:
I'll wait for https://github.com/dusk-network/rusk/pull/3046#discussion_r1895820712 to be resolved first